xarray/tests/test_range_index.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import xarray as xr
+from xarray.indexes import PandasIndex, RangeIndex
+from xarray.tests import assert_allclose, assert_equal, assert_identical
+
+
+def create_dataset_arange(
+    start: float, stop: float, step: float, dim: str = "x"
+) -> xr.Dataset:
+    index = RangeIndex.arange(dim, dim, start, stop, step)
+    return xr.Dataset(coords=xr.Coordinates.from_xindex(index))
+
```

Comment:
this simple test fails:
```
import xarray as xr
from xarray.indexes import RangeIndex

ds = xr.Dataset({"foo": ("x", [1, 2, 3])})
coords = xr.Coordinates.from_xindex(RangeIndex.arange("x", "x", 1.5, 4.5, 1.0))
ds.assign_coords({"x": coords})
```

We are passing
```
{'foo': <xarray.Variable (x: 3)> Size: 24B
array([1, 2, 3]), 'x': <xarray.IndexVariable 'x' (x: 1)> Size: 4B
array(['x'], dtype='<U1')}
```
to `calculate_dimensions`

Reply:
This should be `ds.assign_coords(coords)` and not `ds.assign_coords({"x": coords})` since `coords` is already a mapping object.